keyboards/nuovotaka/freeadrs32_knob/rev1/keymaps/default/keymap.c

```diff
@@ -0,0 +1,62 @@
+                KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I,
+        //|------+--------+--------+--------+--------+-------+--------+--------+--------|
+        KC_1, KC_LSFT, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J,
+        //|------+--------+--------+--------+--------+-------+--------+--------+--------|
+        KC_1, KC_LALT, KC_Z, KC_X, KC_C, KC_V, MO(1), KC_M, KC_ENT
+        //|------+--------+--------+--------+--------+-------+--------+--------+--------|
+    ),
+    [1] = LAYOUT(
+        //,------+--------+--------+--------+--------+-------+--------+--------+--------.
+        S(KC_ESC), S(KC_1), S(KC_2), S(KC_3), S(KC_4), S(KC_5), S(KC_6), S(KC_7), S(KC_8),
+        //|------+--------+--------+--------+--------+-------+--------+--------+--------|
+                S(KC_Q), S(KC_W), S(KC_E), S(KC_R), S(KC_T), S(KC_Y), S(KC_U), S(KC_I),
+        //|------+--------+--------+--------+--------+-------+--------+--------+--------|
+        S(KC_1), KC_LSFT, S(KC_A), S(KC_S), S(KC_D), S(KC_F), S(KC_G), S(KC_H), S(KC_J),
+        //|------+--------+--------+--------+--------+-------+--------+--------+--------|
+        S(KC_1), KC_LALT, S(KC_Z), S(KC_X), S(KC_C), S(KC_V), MO(1), LT(2, S(KC_M)), KC_ENT
+        //|------+--------+--------+--------+--------+-------+--------+--------+--------|
+    ),
+    [2] = LAYOUT(
+        //,------+--------+--------+--------+--------+-------+--------+--------+--------.
+          _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        //|------+--------+--------+--------+--------+-------+--------+--------+--------|
+                   _______, _______, _______, _______, _______, _______, _______, _______,
+        //|------+--------+--------+--------+--------+-------+--------+--------+--------|
+            _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        //|------+--------+--------+--------+--------+-------+--------+--------+--------|
+            _______, _______, _______, _______, _______, _______, _______, _______, _______
+        //|------+--------+--------+--------+--------+-------+--------+--------+--------|
+    )
+};
+
+// clang-format on
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] = {
+        ENCODER_CCW_CW(KC_PAGE_UP, KC_PAGE_DOWN), // Encoder 1
+        ENCODER_CCW_CW(KC_UP, KC_DOWN),           // Encoder 2
+        ENCODER_CCW_CW(KC_LEFT, KC_RIGHT)         // Encoder 3
+    },
+    [1] = {
+        ENCODER_CCW_CW(KC_PAGE_UP, KC_PAGE_DOWN), // Encoder 1
+        ENCODER_CCW_CW(KC_UP, KC_DOWN),           // Encoder 2
+        ENCODER_CCW_CW(KC_LEFT, KC_RIGHT)         // Encoder 3
+    },
+    [2] = {
+        ENCODER_CCW_CW(KC_PAGE_UP, KC_PAGE_DOWN), // Encoder 1
+        ENCODER_CCW_CW(KC_UP, KC_DOWN),           // Encoder 2
+        ENCODER_CCW_CW(KC_LEFT, KC_RIGHT)         // Encoder 3
```

Comment:
Not necessary, but you can use transparency here, so you don't have to explicitly list all of the keycodes. 
```suggestion
        ENCODER_CCW_CW(_______, _______),        // Encoder 1
        ENCODER_CCW_CW(_______, _______),        // Encoder 2
        ENCODER_CCW_CW(_______, _______)         // Encoder 3
    },
    [2] = {
        ENCODER_CCW_CW(_______, _______),        // Encoder 1
        ENCODER_CCW_CW(_______, _______),        // Encoder 2
        ENCODER_CCW_CW(_______, _______)         // Encoder 3
```